File: ErrorBot.js
// ErrorBot integration for Node.js

const https = require('https');
const url = require('url');

class ErrorBot {
  constructor(apiKey, projectName) {
    this.apiKey = apiKey;
    this.projectName = projectName;
    this.endpoint = 'https://errorbot.fyi/api/v1/report';
  }

  init() {
    process.on('uncaughtException', (err) => {
      this.reportError(`Unhandled exception: ${err.message}`, 'error');
      process.exit(1);
    });

    process.on('unhandledRejection', (reason, promise) => {
      this.reportError(`Unhandled rejection at ${promise}: ${reason}`, 'error');
    });
  }

  reportError(message, type = 'error') {
    const errorData = {
      message,
      type,
      project: this.projectName
    };

    const parsedUrl = url.parse(this.endpoint);
    const options = {
      method: 'POST',
      hostname: parsedUrl.hostname,
      path: parsedUrl.path,
      headers: {
        'Content-Type': 'application/json',
        'X-API-Key': this.apiKey
      }
    };

    const req = https.request(options, (res) => {
      let data = '';
      res.on('data', (chunk) => {
        data += chunk;
      });
      res.on('end', () => {
        try {
          const result = JSON.parse(data);
          if (result.success) {
            console.log('Error reported successfully');
          } else {
            console.error(`Failed to report error: ${result.error.message}`);
          }
        } catch (e) {
          console.error('Failed to parse ErrorBot response:', e);
        }
      });
    });

    req.on('error', (error) => {
      console.error('Failed to report error to ErrorBot:', error);
    });

    req.write(JSON.stringify(errorData));
    req.end();
  }
}

// Usage
const errorBot = new ErrorBot('YOUR_API_KEY_HERE', 'Your Project Name');
errorBot.init();

// To manually report an error
// errorBot.reportError('Something went wrong', 'warning');

module.exports = errorBot;

File: error_bot.py
# ErrorBot integration for Python

import sys
import requests
from datetime import datetime

class ErrorBot:
    def __init__(self, api_key, project_name):
        self.api_key = api_key
        self.project_name = project_name
        self.endpoint = 'https://errorbot.fyi/api/v1/report'

    def init(self):
        sys.excepthook = self.handle_exception

    def handle_exception(self, exc_type, exc_value, exc_traceback):
        if issubclass(exc_type, KeyboardInterrupt):
            sys.__excepthook__(exc_type, exc_value, exc_traceback)
            return

        error_message = f"Unhandled exception: {exc_value}"
        self.report_error(error_message)

    def report_error(self, message, error_type='error'):
        error_data = {
            'message': message,
            'type': error_type,
            'project': self.project_name
        }

        try:
            response = requests.post(
                self.endpoint,
                json=error_data,
                headers={
                    'Content-Type': 'application/json',
                    'X-API-Key': self.api_key
                }
            )
            response.raise_for_status()
            result = response.json()
            if result['success']:
                print("Error reported successfully")
            else:
                print(f"Failed to report error: {result['error']['message']}")
        except requests.RequestException as e:
            print(f"Failed to report error to ErrorBot: {e}", file=sys.stderr)

# Usage
error_bot = ErrorBot('YOUR_API_KEY_HERE', 'Your Project Name')
error_bot.init()

# To manually report an error
# error_bot.report_error('Something went wrong', 'warning')